LastChanges-ItayDev

itaycsguy opened the pull request.

itaycsguy committed:
replaced Constant values into static variable values from the relevant class

itaycsguy committed:
added popup frame implementation for image metadata editing

itaycsguy committed:
repaired:
1. profile fields layout
2. profile image border

itaycsguy committed:
added:
1. dropdown notes list
2. google search places - by cities
3. anonymous to Anonymous in Firebase database

itaycsguy committed:
added:
1. ultimative metadata fill box - using google search places API to find a desired location which is the note is found
2. added Special notes to be writter using free text
3. changed customSnackBar colors
4. moved 'Logged in as ${username}' to profile page

itaycsguy committed:
checking user permission - on going

Morag140 commented on app/src/main/java/app/itaycsguy/musiciansaidb/FirebaseDB.kt:
```diff
@@ -32,8 +36,17 @@ class FirebaseDB {
+
+    fun writeTempImg(img : ImageView){
         val userEntry = _firebaseDB.getReference("temp_img")
 //        System.err.println("the drawable object looks like this: ${img.drawable}")
```
comment out can be removed now if not used

Morag140 commented on app/src/main/java/app/itaycsguy/musiciansaidb/GoogleAuth.kt:
```diff
@@ -64,13 +69,40 @@ class GoogleAuth(act : AppCompatActivity) : GoogleApiClient.OnConnectionFailedLi
+                }
+            }
             _userData = map
+//            TODO: make this changing asynchronous:
```
remove all the comment out

Morag140 commented on app/src/main/java/app/itaycsguy/musiciansaidb/MenuActivity.kt:
```diff
@@ -75,7 +92,7 @@ class MenuActivity() : AppCompatActivity(), Parcelable {
         //Get user data from Start Activity
         _user = User(intent.getSerializableExtra("user") as HashMap<String,String>)
-        Toast.makeText(this, "Logged in as ${_user.getUserName()}.", Toast.LENGTH_SHORT).show()
+        // Toast.makeText(this, "Logged in as ${_user.getUserName()}.", Toast.LENGTH_SHORT).show()
```
remove if u dont want it here

Morag140 commented on app/src/main/java/app/itaycsguy/musiciansaidb/MenuActivity.kt:
```diff
@@ -106,16 +123,115 @@ class MenuActivity() : AppCompatActivity(), Parcelable {
 
 
         uploadButton.setOnClickListener {
-            uploadImage()
```
this whole function was changed in my commit so im not sure if this all needed

Morag140 commented on app/src/main/java/app/itaycsguy/musiciansaidb/MenuActivity.kt:
```diff
@@ -106,16 +123,115 @@ class MenuActivity() : AppCompatActivity(), Parcelable {
+                                    Toast.makeText(currAct, "Could not meet an updating", Toast.LENGTH_LONG).show()
+                                }
+                            } else {
+                                Toast.makeText(currAct, "All fields are required to be filled.", Toast.LENGTH_LONG).show()
```
why are they all required? if he doesnt know who wrote it or where was it or w.e? do we want to really enforce them all?

Morag140 commented on app/src/main/res/values/strings.xml:
```diff
@@ -18,4 +18,252 @@
+        <item>B</item>
+        <item>Special Note</item>
+    </string-array>
+    <!--<string-array name="countries_array">-->
```
why is this all here if its commented out?

Morag140 reviewed (changes requested):
BUG BUG BUG : for some reason the edit meta data "location" just opens and closes...nothing else (cant work with it) and because it doesnt work and you FORCE all fields to be filled then the entire meta data feature doesn't work for me. also i would reccomend not to force all fields but let them fill what they know (as far as i remember the important reason this application is made is because they upload notes of chords they don't know who was the writer so its kinda stupid to force them to put one)

look at my comments...dont forget to rebase this after merging my pull request.
make sure it all works...and if you are not sure let me check it again before u merge it.

itaycsguy committed:
Determined to live the way google's users can connect without keeping their lifecycle permission which they maybe achieve along their being into the application

itaycsguy committed:
.

itaycsguy committed:
fixed profile activity layout!

itaycsguy committed:
checked!

itaycsguy committed:
Handled profile,registration and sign-up flexible layout

itaycsguy committed:
Handled profile,registration and sign-up flexible layout - version 2

itaycsguy committed:
Added - next flow: picking an image-> append its metadata -> upload image

Morag140 committed:
Itay's Last Changes Branch + Bug fixes.

* This Commit contains all of the work done by Itay plus various bug fixes.

* Location setting for an image meta data is currently not forced on the user since it is not working for all versions of this app.

* Some Type fixes.

* Made sure the Image flow works and does not crush on mid work canceling
* Made sure that on errors the Image is reset to the default.
* Optimized imports on MenuActivity
* Gradle build- moved all google to be of same version and close to each other.

This was tested on Android api 28 emulator and Samsung 7 (mine) for full Upload Flow (did not check login and registration flows)
